osf/models/mixins.py

```diff
@@ -805,7 +805,8 @@ def set_subjects(self, new_subjects, auth, add_log=True):
 
         :return: None
         """
-        self.check_subject_perms(auth)
+        if auth and not (hasattr(self, 'has_permission') and self.has_permission(auth.user, ADMIN)):
```

Comment:
My fault, I wasn't clear with my last suggestion, when I said compare to `Node.set_privacy`.  

I just meant check if `auth` is present before doing permission checks. 

So just 
````
if auth:
      self.check_subject_perms(auth)